core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -397,7 +397,7 @@ private boolean isCommaJoin(Join join) {
   public Result visit(Correlate e) {
     final Result leftResult =
         visitInput(e, 0)
-            .resetAlias(e.getCorrelVariable(), e.getRowType());
+            .resetAlias(e.getCorrelVariable(), e.getInput(0).getRowType());
```

Comment:
this seems like a pretty big bug, I wonder how it wasn't discovered sooner
Is it because for correlated queries the type of e is often the same as the type of e.getInput(0)?

Reply:
It may be that fewer cases are used. Correlate is similar to Join, with leftNode and rightNode, among which e.getInput(0) is equivalent to leftNode. Therefore the RowType of e.getInput(0) is a subset of Correlate.